x-pack/plugins/cross_cluster_replication/__jest__/client_integration/test_helpers.js

```diff
@@ -0,0 +1,198 @@
+
+        let button;
+        if (action === 'delete') {
+          button = findTestSubject(tableCellActions, 'ccrAutoFollowPatternListDeleteActionButton');
+        } else if (action === 'edit') {
+          findTestSubject(tableCellActions, 'ccrAutoFollowPatternListEditActionButton');
+        }
+
+        if (!button) {
+          throw new Error(`Button for action "${action}" not found.`);
+        }
+
+        button.simulate('click');
+      };
+
+      const clickAutoFollowPatternAt = (index = 0) => {
+        const autoFollowPatternLink = findTestSubject(rows[index].reactWrapper, 'ccrAutoFollowPatternListPatternLink');
+        autoFollowPatternLink.simulate('click');
+      };
+
+      return {
+        selectAutoFollowPatternAt,
+        clickBulkDeleteButton,
+        clickConfirmModalDeleteAutoFollowPattern,
+        clickRowActionButtonAt,
+        clickAutoFollowPatternAt
+      };
+    }
+  };
+
+  return userActions[section]();
+};
+
+export { nextTick, getRandomString, findTestSubject } from '../../../../test_utils';
+
+export const initTestBed = (component, props = {}, options) => {
+  const testBed = registerTestBed(component, {}, ccrStore)(props, options);
+  const getUserActions = initUserActions(testBed);
+
+  return {
+    ...testBed,
+    getUserActions,
+  };
+};
+
+export const mockServerResponses = server => {
+  const mockResponse = (defaultResponse, response) => ([
+    200,
+    { 'Content-Type': 'application/json' },
+    JSON.stringify({ ...defaultResponse, ...response }),
+  ]);
+
+  const mockLoadFollowerIndices = (response) => {
+    const defaultResponse = { indices: [] };
+
+    server.respondWith('GET', /api\/cross_cluster_replication\/follower_indices/,
+      mockResponse(defaultResponse, response));
+  };
+
+  const mockLoadAutoFollowPattern = (response) => {
+    const defaultResponse = { patterns: [] };
+
+    server.respondWith('GET', /api\/cross_cluster_replication\/auto_follow_patterns/,
+      mockResponse(defaultResponse, response)
+    );
+  };
+
+  const mockDeleteAutoFollowPattern = (response) => {
+    const defaultResponse = { errors: [], itemsDeleted: [] };
+
+    server.respondWith('DELETE', /api\/cross_cluster_replication\/auto_follow_patterns/,
+      mockResponse(defaultResponse, response)
+    );
+  };
+
+  const mockAutoFollowStats = (response) => {
+    const defaultResponse = {
+      numberOfFailedFollowIndices: 0,
+      numberOfFailedRemoteClusterStateRequests: 0,
+      numberOfSuccessfulFollowIndices: 0,
+      recentAutoFollowErrors: [],
+      autoFollowedClusters: [{
+        clusterName: 'new-york',
+        timeSinceLastCheckMillis: 13746,
+        lastSeenMetadataVersion: 22
+      }]
+    };
+
+    server.respondWith('GET', /api\/cross_cluster_replication\/stats\/auto_follow/,
+      mockResponse(defaultResponse, response)
+    );
+  };
+
+  mockLoadFollowerIndices();
+  mockLoadAutoFollowPattern();
+  mockAutoFollowStats();
```

Comment:
I'm having a hard time understanding why this help needs to call these functions immediately. Can we make the consumer of this helper responsible for calling the mocking functions that it needs instead of calling them here? I think that will make the code more explicit and easier to follow.

Reply:
So the `mockServerResponses()` _is_ the helper that will mock all server requests needed. It is called once, on before each, to make things much terser in the tests (http request have been handled, we can move on). 
What I decided to do, is export helper methods from it to be able to override the mock whenever needed.
So it first sets the required mock for the requests and export helpers for overrides. I can add a comment or maybe give a better name to the function.